offload/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -205,13 +205,16 @@ Expected<std::string> getTargetTripleAndFeatures(hsa_agent_t Agent) {
       return Status;
 
     llvm::StringRef TripleTarget(ISAName.begin(), Length);
-    if (TripleTarget.consume_front("amdgcn-amd-amdhsa"))
-      Target = TripleTarget.ltrim('-').rtrim('\0').str();
-    return HSA_STATUS_INFO_BREAK;
+    if (TripleTarget.consume_front("amdgcn-amd-amdhsa")) {
+      auto Target = TripleTarget.ltrim('-').rtrim('\0').str();
+      if (Target.find("generic") != std::string::npos)
+        Targets.push_back(Target);
+      else
+        Targets[0] = Target;
```

Comment:
Expect only one specific ISA. Code is a bit overkill for today's implementation where HSA lists specific ISA upfront and thus we really don't need to IF branch. I just didn't like to blindly assume that ordering.

Reply:
I don't think this needs any extra handling. You can just collect all HSA ISAs (no matter they are generic or not) in this function, and then check if any of them is compatible. I don't think there needs to be anything specific at OpenMP level.

Reply:
If you are in the camp of "leaving the rest to PluginManager to handle" for the foreseeable future, that would be the case. I'm not in that camp at this point and thus am seeking whether there is a good away to address that soon enough.

Reply:
Revised code uploaded. New code returns SpecificTarget (same as before). All target ISAs are push_back-ed to the newly added argument Targets.